docs: tidying up more guides (#147)
NordicSemiconductor/asset-tracker-cloud-docs: docs/guides/Versioning.rst

.. note::

    A commit message with different format will not trigger a release.

Patch release (fix)
===================

A fix is a change to the implementation, which removes a bug without changing the public API (method names and signatures, exports) of the library.
The consumer of the library does not need to change their implementation when installing the fixed version.

> To release a fix (update the version from ``2.3.4`` to ``2.3.5``), use the type ``fix`` in the commit message:

Reviewer: ```suggestion
To release a fix (that updates the version from ``2.3.4`` to ``2.3.5``), use the type ``fix`` in the commit message:
```

.. parsed-literal::
   :class: highlight

    fix: *commit message*

You can optionally add a scope, which is a lowercase string that is used to identify the affected component.

.. parsed-literal::
   :class: highlight

    fix(*scope*): *commit message*

Scopes are useful when working with a large project.

Feature release (feat)
======================

A feature is a change to the implementation of a library which adds additional functionality without changing the public API of the library.

You may add new exports, methods and extra method arguments that are optional, but the consumer of the library does not need to change their implementation when installing the library with the new feature.

> To release a feature (update the version from ``2.3.4`` to ``2.4.0``), use the type ``feat`` in the commit message:

Reviewer: ```suggestion
To release a feature (that updates the version from ``2.3.4`` to ``2.4.0``), use the type ``feat`` in the commit message:
```
```suggestion
To release a feature (update the version from ``2.3.4`` to ``2.4.0``), use the type ``feat`` in the commit message:
```

Breaking Release (BREAKING CHANGE:)
===================================

Whenever the public API of the library changes in a way that requires the consumer to update their implementation, you must mark the change as breaking.
This can happen in the following conditions:

* Method names and arguments are renamed.
* Previously optional arguments become mandatory.
* Arguments are removed.
* Classes are renamed.

Even if the change is very small, as soon as you break the library's contract (its public API), it is a breaking change according to `Semantic Versioning 2.0.0  <https://semver.org/#spec-item-8>`_.

> To do a breaking release (update the version from ``2.3.4`` to ``3.0.0``), use the text :code:`BREAKING CHANGE:` (followed by a space of two new lines) in the commit message:

Reviewer: ```suggestion
To do a breaking release (that updates the version from ``2.3.4`` to ``3.0.0``), use the text :code:`BREAKING CHANGE:` (followed by a space of two new lines) in the commit message:
```
```suggestion
To do a breaking release (update the version from ``2.3.4`` to ``3.0.0``), use the text :code:`BREAKING CHANGE:` (followed by a space of two new lines) in the commit message:
```
